### PR TITLE
✨ Add a script: login->sync->syncRecords

### PR DESCRIPTION
**Pull request by grdryn**
This doesn't yet work.

### CODE REVIEW

File: scripts/sync-workflow.js [hunk 1]
```diff
@@ -58,3 +61,59 @@ function login(resolve, reject) {
...
+      url: `${argv.app}/cloud/workorders`,
+      body: reqBody,
+      json: true
+    }, (err, httpResponse, resBody) => {
+      console.warn('Got a reply');
+      if (err) {
+        console.error(err);
+        return reject(err);
+      }
+      console.warn(`#########################${util.inspect(resBody)}#########################`);
```
Comment (grdryn) [inline scripts/sync-workflow.js]: `resBody` here is the string 'ok', but I'm expecting something like an object with an updates property.

File: scripts/sync-workflow.js [hunk 2]
```diff
@@ -58,3 +61,59 @@ function login(resolve, reject) {
...
+    lr.actStart('EMPTY SYNC');
+
+    const myRequest = request.defaults({
+      headers: requestBodyUtils.getSyncRequestHeaders({
+        'X-FH-cuid': clientIdentifier,
+        'X-FH-sessionToken': loginResult.sessionToken
+      })
+    });
+
+    const reqBody = requestBodyUtils.getSyncRequestBody({
```
Comment (grdryn) [inline scripts/sync-workflow.js]: Most of the stuff here I got from inspecting requests sent from the browser. Maybe some of it is not needed.

File: scripts/sync-workflow.js [hunk 3]
```diff
@@ -58,3 +61,59 @@ function login(resolve, reject) {
...
+      dataset_id: 'workorders',
+      query_params: {
+        "filter": {
+          "key": "assignee",
+          "value": "rkX1fdSH"
+        }
+      },
+      dataset_hash: '833c31caee5b87f4cbf9baeab0f5dce6814467ee',
+      meta_data: {
```
Comment (grdryn) [inline scripts/sync-workflow.js]: This bit I copied from @aidenkeating. I think in the browser requests that I was looking at, `meta_data` was set to `null`. I've tried both ways, and it doesn't seem to have an effect on the result.

File: scripts/sync-workflow.js [hunk 4]
```diff
@@ -58,3 +61,59 @@ function login(resolve, reject) {
     return resolve(resBody);
   });
 }
+
+function emptySync(loginResult) {
+  return function(resolve, reject) {
```
Comment (grdryn) [inline scripts/sync-workflow.js]: There might be a better way to structure this (and L38), suggestions welcome.

File: scripts/sync-workflow.js [hunk 5]
```diff
@@ -58,3 +61,59 @@ function login(resolve, reject) {
...
+
+    console.log(reqBody);
+    console.warn('About to post request');
+    myRequest.post({
+      url: `${argv.app}/cloud/workorders`,
+      body: reqBody,
+      json: true
+    }, (err, httpResponse, resBody) => {
+      console.warn('Got a reply');
+      if (err) {
```
Comment (grdryn) [inline scripts/sync-workflow.js]: Maybe using bluebird to promisify `request.post` would make this simpler.

**Review by grdryn — COMMENTED** [review]
The point of this PR is more to try to get a sync request to the raincatcher app to work, rather than something that should get merged.

It's not currently working, so I'm hoping you guys can point out something obvious that's wrong, @aidenkeating @paolobueno. All feedback welcome!

**Comment by aidenkeating** [comment]
@grdryn I couldn't set myself as a reviewer, probably because of permissions, but I am taking a look now.

File: scripts/sync-workflow.js [hunk 6]
```diff
@@ -58,3 +61,59 @@ function login(resolve, reject) {
...
+        "appkey": "b916e86a5366aff35dabab86654f4c7f41f82b0a",
+        "projectid": "zf47cmeatrf5iqvypq5tilq7",
+        "connectiontag": "0.0.1"
+      }
+    });
+
+    console.log(reqBody);
+    console.warn('About to post request');
+    myRequest.post({
+      url: `${argv.app}/cloud/workorders`,
```
Comment (grdryn) [inline scripts/sync-workflow.js]: Thanks @aidenkeating for pointing out that the path is wrong here and should be instead `/mbaas/sync/workorders`.

I was hitting this: https://github.com/feedhenry-raincatcher/raincatcher-demo-cloud/blob/master/lib/app.js#L58..L60

Which I don't understand the purpose of :P

**Comment by grdryn** [comment]
I've made some changes to this, and it works much better now. I think it's a sufficient starting point! :)

**Comment by grdryn** [comment]
@paolobueno Do you think it would be okay to merge this? We can change it and build on it as needed

**Review by paolobueno — APPROVED** [review]

**Comment by paolobueno** [comment]
@grdryn sorry, missed the GH notification mail earlier!

**Pull request merged**